backend/protondrive/protondrive.go

```diff
@@ -0,0 +1,1040 @@
+		Used:  &used,
+		Free:  &free,
+	}
+
+	return usage, nil
+}
+
+// ------------------------------------------------------------
+
+// Fs returns the parent Fs
+func (o *Object) Fs() fs.Info {
+	return o.fs
+}
+
+// Return a string version
+func (o *Object) String() string {
+	if o == nil {
+		return "<nil>"
+	}
+	return o.remote
+}
+
+// Remote returns the remote path
+func (o *Object) Remote() string {
+	return o.remote
+}
+
+// Hash returns the hashes of an object
+func (o *Object) Hash(ctx context.Context, t hash.Type) (string, error) {
+	return "", hash.ErrUnsupported
+}
+
+// Size returns the size of an object in bytes
+func (o *Object) Size() int64 {
+	if o.fs.opt.ReportOriginalSize {
+		// if ReportOriginalSize is set, we will generate an error when the original size failed to be parsed
+		// this is crucial as features like Open() will need to use the proper size to operate the seek/range operator
+		if o.originalSize != nil {
+			return *o.originalSize
+		}
+
+		fs.Errorf(o, "Original size should exist")
+	}
+	return o.size
+}
+
+// ModTime returns the modification time of the object
+//
+// It attempts to read the objects mtime and if that isn't present the
+// LastModified returned in the http headers
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	return o.modTime
+}
+
+// SetModTime sets the modification time of the local fs object
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+// Storable returns a boolean showing whether this object storable
+func (o *Object) Storable() bool {
+	return true
+}
+
+// Open opens the file for read.  Call Close() on the returned io.ReadCloser
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (io.ReadCloser, error) {
+	// download and decrypt the file
+	var reader io.ReadCloser
+	var fileSystemAttrs *protonDriveAPI.FileSystemAttrs
+	var sizeOnServer int64
+	var err error
+	if err = o.fs.pacer.Call(func() (bool, error) {
+		reader, sizeOnServer, fileSystemAttrs, err = o.fs.protonDrive.DownloadFileByID(ctx, o.id)
+		return shouldRetry(ctx, err)
+	}); err != nil {
+		return nil, err
+	}
+
+	if fileSystemAttrs != nil {
+		o.originalSize = &fileSystemAttrs.Size
+		o.modTime = fileSystemAttrs.ModificationTime
+	} else {
+		fs.Debugf(o, "fileSystemAttrs is nil: using fallback size")
+		o.originalSize = &sizeOnServer
+		o.size = sizeOnServer
+	}
+
+	fs.FixRangeOption(options, *o.originalSize)
+	var offset, limit int64 = 0, -1
+	for _, option := range options { // if the caller passes in nil for options, it will become array of nil
+		switch x := option.(type) {
+		case *fs.SeekOption:
+			offset = x.Offset
+		case *fs.RangeOption:
+			offset, limit = x.Decode(o.Size())
+		default:
+			if option.Mandatory() {
+				fs.Logf(o, "Unsupported mandatory option: %v", option)
+			}
+		}
+	}
+
+	retReader := io.NopCloser(reader) // the NewLimitedReadCloser will deal with the limit
+
+	// deal with offset
```

Comment:
I believe that with the `noMultiThreading`, the file will only be downloaded once. But I am not certain what are the use cases for `Seek`. 

Due to the way the file is encrypted, as explained and discussed [here](https://github.com/rclone/rclone/pull/7093#issuecomment-1636737655), I don't have a better way to support seeking in this backend. 

But never the less, the code now is very efficient, as the download is done in a way that
- we decrypt and buffer data of 8 blocks 
- the reader can perform read
- when the buffer runs out of data, we decrypt and buffer data up to next 8 blocks 
- repeat until EOF 

For `Seek`, it's just reading but discarding the result, it's a bit wasteful, but at least it's memory efficient and the file is downloaded once only!